scripts/makehtml.py

```diff
@@ -146,7 +146,7 @@ def parse_device(svdfile):
             # Bodge to prevent /0 when there are no fields in a register
             if register_fields_total == 0:
                 register_fields_total = 1
-            registers[roffset] = {"name": rname,
+            registers[rname] = {"name": rname,
```

Comment:
```suggestion
            registers[(roffset, rname)] = {"name": rname,
```

I had to check to remember what this is used for, but it's inside `makehtml.template.html`, where the registers are iterated over by sorted key order, so previously they were in register-offset order but now they'd be in register-alphabetical order. If we make this key (offset, name), we preserve the sort order but allow the overlapping registers to still appear.

I think it would be good to also change `makehtml.template.html` L121 to something like `{% for _, register in peripheral.registers|dictsort %}` (changing `roffset` to `_`) to make it clear that the key isn't used later, since it's now a tuple.

Finally the indentation of the fields below needs updating since this line changes length.

Reply:
Good catch, I thought registers were inserted by their roffset already, but I missed the sorting function.